### ISSUE TITLE
_pm_ispartition() heulistic miss some block devices

### ISSUE THREAD

**Issue by moriwaka** [performancecopilot/pcp#45]
In src/pmdas/linux/proc_partitions.c, _pm_ispartition() use heulistic to detect partitions, but this misses some block devices like "sr0", "scd0", "fd0" etc. 
I suggest using sysfs.  /sys/block/* listing all block devices.

Thanks,

**Comment by natoscott** [comment]
I believe this was done purposefully, because floppy disk, CDROM, and DVD devices are not typically of interest in general purpose performance analysis tools.  Do you have a need for the per-partition values from those devices?

**Comment by moriwaka** [comment]
I noticed this by difference of pmiostat and iostat outputs.  I agree with you, CD,DVD,FDs are not interested in almost all use case. To now, I don't have real needs nor problems.

I thought that new devices' or some special drivers' naming rules are uncontrollable. And proper block device detection method can be help for minor devices and users.

Thanks,

**Comment by natoscott** [comment]
> I thought that new devices' or some special drivers' naming rules are uncontrollable.
> And proper block device detection method can be help for minor devices and users.

Yes, this could certainly be done and would be an improvement I think.  We would need to ensure the current code remains though, as it will need to run on older kernels lacking sysfs support.  Would you be interested in sending through a patch?

**Comment by moriwaka** [comment]
Yes, I'll try to make a patch.

**Comment by kmcdonell** [comment]
Do we need to keep this (eleven year old!) issue open?  I'm not sure that /sys/block is any more reliable than the current heuristic, as mine contains 50 entries ... 44 of them are loop<N> and one is a DVD, and there are only 5 real "disks" that warrant disk.dev.* metrics, so we'd still need some filtering heuristic even with /sys/block.
```
kenj@bozo:~/src/pcp$ ls /sys/block
loop0   loop13  loop18  loop22  loop27  loop31  loop36  loop40  loop6  sdb
loop1   loop14  loop19  loop23  loop28  loop32  loop37  loop41  loop7  sdc
loop10  loop15  loop2   loop24  loop29  loop33  loop38  loop42  loop8  sdd
loop11  loop16  loop20  loop25  loop3   loop34  loop39  loop43  loop9  sde
loop12  loop17  loop21  loop26  loop30  loop35  loop4   loop5   sda    sr0
```
Unless there is a real problem here, or a patch to consider, I vote for closing this issue.
